File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3101_count-alternating-subarrays/solutions/c/solution.c
#include <stdio.h>

long long countAlternatingSubarrays(int* nums, int numsSize) {
    long long total = 0;
    int length = 1;
    
    for (int i = 1; i < numsSize; i++) {
        if (nums[i] != nums[i - 1]) {
            length++;
        } else {
            total += (long long)length * (length + 1) / 2;
            length = 1;
        }
    }
    
    total += (long long)length * (length + 1) / 2;
    
    return total;
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3101_count-alternating-subarrays/solutions/java/solution.java
class Solution {
    public long countAlternatingSubarrays(int[] nums) {
        long totalAlternating = 0;
        int start = 0;
        
        while (start < nums.length) {
            int length = 1;
            while (start + length < nums.length && nums[start + length] != nums[start + length - 1]) {
                length++;
            }
            totalAlternating += ((long) length * (length + 1)) / 2;
            start += length;
        }
        
        return totalAlternating;
    }
}
